include/net/flow_offload.h

```diff
@@ -118,6 +120,49 @@ void flow_rule_match_enc_opts(const struct flow_rule *rule,
 void flow_rule_match_ct(const struct flow_rule *rule,
 			struct flow_match_ct *out);
 
+void flow_rule_match_unmasked_key_meta(const struct flow_rule *rule,
+				       struct flow_match_meta *out);
```

Comment:
@dsatish 
All these new functions defined are later exported. Can we have a common interface with additional argument to decide the final method like:
```
void flow_rule_match_unmasked_key(const struct flow_rule *rule,
				       struct flow_match_meta *out, const uint32_t type);
```
That way only one function need to be exported.

Reply:
Agree but this way we have to change the type from struct flow_match_meta * to void *, as this input structure will change based on type. We inherited this kind of exporting methods from existing code only for parsing masked key.

Reply:
I see. Then this is coding policy "code like the code around" which is followed by most open source communities. Looks fine then.

Reply:
Yes.